File: src/main/java/com/pauljoda/modularsystems/core/api/nei/NEIAddonConfig.java
package com.pauljoda.modularsystems.core.api.nei;

import codechicken.nei.api.API;
import codechicken.nei.api.IConfigureNEI;
import codechicken.nei.recipe.TemplateRecipeHandler;
import com.pauljoda.modularsystems.core.ModularSystems;
import com.pauljoda.modularsystems.core.api.nei.machines.RecipeHandlerBlockValues;
import com.pauljoda.modularsystems.core.api.nei.machines.RecipeHandlerCrusher;
import com.pauljoda.modularsystems.core.api.nei.machines.RecipeHandlerFurnace;
import com.pauljoda.modularsystems.core.api.nei.storage.GuiStorageCraftingHandler;
import com.pauljoda.modularsystems.core.lib.Reference;
import com.pauljoda.modularsystems.storage.gui.GuiStorageCore;
import net.minecraft.util.StatCollector;

/**
 * This will be called automatically when NEI is loaded. Must be called this exactly for NEI to find it
 *
 * Make sure if you create a handler to register it here
 */
public class NEIAddonConfig implements IConfigureNEI {
    @Override
    public void loadConfig() {

        registerHandler(new RecipeHandlerCrusher());
        registerHandler(new RecipeHandlerFurnace());
        registerHandler(new RecipeHandlerBlockValues());

        API.registerGuiOverlay(GuiStorageCore.class, "crafting");
        API.registerGuiOverlayHandler(GuiStorageCore.class, new GuiStorageCraftingHandler(180, 162), "crafting");

        //By setting this here, we can let the rest of the mod know NEI is installed
        ModularSystems.nei = new NEICallback();
    }

    @Override
    public String getName() {
        return StatCollector.translateToLocal("modularsystems.nei.name");
    }

    @Override
    public String getVersion() {
        return Reference.VERSION;
    }

    /**
     * Little helper method to register the handlers
     * @param handler The handler to register
     */
    private static void registerHandler(TemplateRecipeHandler handler) {
        API.registerRecipeHandler(handler);
        API.registerUsageHandler(handler);
    }
}

File: src/main/scala/com/teambr/modularsystems/core/api/nei/NEIAddonConfig.scala
package com.teambr.modularsystems.core.api.nei

import codechicken.nei.api.{ API, IConfigureNEI }
import codechicken.nei.recipe.TemplateRecipeHandler
import com.teambr.modularsystems.core.ModularSystems
import com.teambr.modularsystems.core.lib.Reference
import net.minecraft.util.StatCollector

/**
 * This file was created for Modular-Systems
 *
 * Modular-Systems is licensed under the
 * Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License:
 * http://creativecommons.org/licenses/by-nc-sa/4.0/
 *
 * @author Paul Davis <pauljoda>
 * @since August 05, 2015
 */
class NEIAddonConfig extends IConfigureNEI {
    override def loadConfig() : Unit = {

        //By setting this here, we can let the rest of the mod know NEI is installed
        ModularSystems.nei = new NEICallback
    }

    override def getName : String = StatCollector.translateToLocal("modularsystems.nei.name")

    override def getVersion : String = Reference.VERSION

    /**
     * Little helper method to register the handlers
     * @param handler The handler to register
     */
    private def registerHandler(handler : TemplateRecipeHandler) {
        API.registerRecipeHandler(handler)
        API.registerUsageHandler(handler)
    }
}
